modules/bench/src/main/clojure/xtdb/bench/auctionmark.clj

```diff
@@ -666,4 +665,4 @@
                              :duration duration
                              :freq (Duration/ofMillis (* 0.2 (.toMillis duration)))
                              :job-task {:t :call, :transaction :index-item-status-groups, :f (b/wrap-in-catch index-item-status-groups)}}]}
-            (when sync {:t :call, :f #(tu/then-await-tx (:sut %))})])}))
+            (when sync {:t :call, :f #(catchup (:sut %))})])}))
```

Comment:
Primarily updated to remove the dependency on `test-util` (caused some issues when building the shadow JAR, since test-util doesn't live under any of the modules, as far as I can tell). File was recently updated to add/use `catchup` - though there IS a difference here:

```clojure
(defn then-await-tx
  (^TransactionKey [node]
   (then-await-tx (latest-submitted-tx node) node nil))

  (^TransactionKey [tx node]
   (then-await-tx tx node nil))

  (^TransactionKey [tx node timeout]
   (idx/await-tx tx node timeout)))
```

Reply:
Rather than catchup:

```clojure
(defn catchup [node]
  (let [{:keys [latest-completed-tx]} (xt/status node)]
    (loop [last-tx latest-completed-tx]
      (Thread/sleep 100)
      (let [{:keys [latest-completed-tx]} (xt/status node)]
        (when-not (= last-tx latest-completed-tx)
          (recur latest-completed-tx))))))
```

Might explain our query 'memory leaks' - might not be letting things finish properly.